### PR TITLE
Fix up Zora River logic

### PR DESCRIPTION
**Pull request by XenoWars**
- Adds a new region in Overworld.yml for Zora River Front
- Accounts for accessing Zora River as either age from all entrances
- Moves the Tree GS to Zora River Front
- Reworks logic for some of the other checks to account for both ages
- Removes ranged requirement for the Ladder GS as it can be jump slashed
- Lowered Bridge GS to Hookshot
- Adds two new macros, for diving deeper
- Adds the can_dive_big macro to Tektite grotto to cut on space

### CODE REVIEW

File: data/oot/macros.yml [hunk 1]
```diff
@@ -37,3 +37,5 @@
 "has_tunic_zora": "is_adult && has(TUNIC_ZORA)"
 "has_iron_boots": "is_adult && has(BOOTS_IRON)"
 "has_hover_boots": "is_adult && has(BOOTS_HOVER)"
+"can_dive_small": "has(SCALE) || has_iron_boots"
+"can_div_big": "has(SCALE, 2) || has_iron_boots"
```
Comment (Nax) [inline data/oot/macros.yml:41]: Typo here

Comment (XenoWars) [reply data/oot/macros.yml:41]: oops

File: data/oot/world/overworld.yml [hunk 2]
```diff
@@ -289,21 +290,27 @@
...
+"Zora River":
+  exits:
+    "Zora River Front": "true"
     "Zora Domain": "can_play(SONG_ZELDA)"
+    "Lost Woods": "can_dive_small"
   locations:
     "Zora River Bean Seller": "is_child"
-    "Zora River HP Pillar": "true"
-    "Zora River HP Platform": "true"
+    "Zora River HP Pillar": "is_child || has_hover_boots"
```
Comment (Celestial-Panda) [inline data/oot/world/overworld.yml:306]: This (the pillar) can also be reached with beans according to main OOTR logic

File: data/oot/world/overworld.yml [hunk 3]
```diff
@@ -289,21 +290,27 @@
...
   exits:
     "Hyrule Field": "true"
+    "Zora River": "is_adult || (is_child && has_explosives)"
+  locations:
+    "Zora River GS Tree": "is_child"
+"Zora River":
+  exits:
+    "Zora River Front": "true"
     "Zora Domain": "can_play(SONG_ZELDA)"
+    "Lost Woods": "can_dive_small"
```
Comment (Celestial-Panda) [inline data/oot/world/overworld.yml:303]: The way back should also be added in Lost Woods

Comment (Nax) [reply data/oot/world/overworld.yml:303]: It is I think, line 155

**Pull request merged**